ansible/roles/iroha-docker/tasks/config-gen.yml

```
@@ -1,3 +1,89 @@
+- name: Check iroha
+  stat:
+    path: "{{ iroha_deploy_dir }}/conf/{{ item.human_hostname }}/genesis.block"
+  loop: "{{ iroha_nodes }}"
+  register: check_iroha_register
+  tags: ["iroha-docker", "iroha-config-gen"]
+
+- block:
+  - name: set empty iroha_new_nodes variable
+    set_fact:
+      iroha_new_nodes: []
+
+  - name: set iroha_new_nodes variable
+    set_fact:
+      iroha_new_nodes: "{{ iroha_new_nodes + [item.item] }}"
+    loop: "{{ check_iroha_register.results }}"
+    when: not item.stat.exists
+
+  - name: set iroha_all_new_nodes variable
+    set_fact:
+      iroha_all_new_nodes: "{{ (iroha_all_new_nodes | default([])) + hostvars[item]['iroha_new_nodes'] }}"
+    loop: "{{ ansible_play_hosts }}"
+    run_once: yes
+
+  - name: New nodes
+    debug:
+      var: iroha_all_new_nodes
+    run_once: yes
+
+  - name: set iroha_old_nodes variable
+    set_fact:
+      iroha_old_nodes: "{{ iroha_nodes | difference(iroha_new_nodes) }}"
+
+  - name: set node_change variable
+    set_fact:
+      node_change: true
```

or you can leave only one task, I did not test it, but is should work 
```suggestion
      node_change: "{{ iroha_all_new_nodes and (( iroha_all_new_nodes | length ) < ( iroha_all_nodes | length )) }}"
```